cmd/tailscaled: nil panic if iptables command not found

Noticed while testing something else.
```
onPortUpdate(port=43435, network=udp6)
router: using firewall mode pref
router: default choosing iptables
UpdateMagicsockPort(port=43435, network=udp6) failed: %!w(*fmt.wrapError=&{could not setup netfilter: could not create new netfilter: exec: "iptables": executable file not found in $PATH 0xc0000ac1a0})
onPortUpdate(port=35190, network=udp4)
magicsock: disco key = ....
Creating WireGuard device...
Bringing WireGuard device up...
Bringing router up...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x86a000]

goroutine 136 [running]:
tailscale.com/util/linuxfw.(*iptablesRunner).HasIPV6(0xc0001d4dd0?)
        tailscale.com/util/linuxfw/iptables_runner.go:158
tailscale.com/wgengine/router.(*linuxRouter).getV6Available(...)
        tailscale.com/wgengine/router/router_linux.go:712
tailscale.com/wgengine/router.(*linuxRouter).addrFamilies(...)
        tailscale.com/wgengine/router/router_linux.go:1108
tailscale.com/wgengine/router.(*linuxRouter).delIPRules(0xc0001d4dd0)
        tailscale.com/wgengine/router/router_linux.go:1327 +0x5a
tailscale.com/wgengine/router.(*linuxRouter).addIPRules(0xc0001d4dd0)
        tailscale.com/wgengine/router/router_linux.go:1124 +0x25
tailscale.com/wgengine/router.(*linuxRouter).Up(0xc0001d4dd0)
        tailscale.com/wgengine/router/router_linux.go:317 +0xee
tailscale.com/wgengine.NewUserspaceEngine(0xc000344050, {{0x134f600, 0xc000534000}, 0x0, {0x1349188, 0xc0001d4dd0}, {0x13490e0, 0xc000344050}, 0x0, 0xc0000f49a0, ...})
        tailscale.com/wgengine/userspace.go:491 +0x15cc
main.tryEngine(0xc000344050, 0xc000233c08, {0x115c3f6, 0xa})
        tailscale.com/cmd/tailscaled/tailscaled.go:742 +0x7e8
main.createEngine(0xc000344050, 0xc000233c08)
        tailscale.com/cmd/tailscaled/tailscaled.go:639 +0x192
main.getLocalBackend({0x0?, 0x0?}, 0xc000344050, {0x1c, 0x76, 0x92, 0x57, 0xaf, 0xaa, 0xfe, ...}, ...)
        tailscale.com/cmd/tailscaled/tailscaled.go:525 +0x11a
main.startIPNServer.func2()
        tailscale.com/cmd/tailscaled/tailscaled.go:481 +0x26b
created by main.startIPNServer in goroutine 1
```

--- a/util/linuxfw/nftables_runner.go
+++ b/util/linuxfw/nftables_runner.go
@@ -592,9 +592,23 @@ func New(logf logger.Logf, prefHint string) (NetfilterRunner, error) {
 	mode := detectFirewallMode(logf, prefHint)
 	switch mode {
 	case FirewallModeIPTables:
-		return newIPTablesRunner(logf)
+		// Note that we don't simply return an newIPTablesRunner here because it
+		// would return a `nil` iptablesRunner which is different from returning
+		// a nil NetfilterRunner.
+		ipr, err := newIPTablesRunner(logf)
+		if err != nil {
+			return nil, err
+		}
+		return ipr, nil
 	case FirewallModeNfTables:
-		return newNfTablesRunner(logf)
+		// Note that we don't simply return an newNfTablesRunner here because it
+		// would return a `nil` nftablesRunner which is different from returning
+		// a nil NetfilterRunner.
+		nfr, err := newNfTablesRunner(logf)
+		if err != nil {
+			return nil, err
+		}
+		return nfr, nil
 	default:
 		return nil, fmt.Errorf("unknown firewall mode %v", mode)
 	}

--- a/wgengine/userspace.go
+++ b/wgengine/userspace.go
@@ -374,7 +374,7 @@ func NewUserspaceEngine(logf logger.Logf, conf Config) (_ Engine, reterr error) 
 		e.logf("onPortUpdate(port=%v, network=%s)", port, network)
 
 		if err := e.router.UpdateMagicsockPort(port, network); err != nil {
-			e.logf("UpdateMagicsockPort(port=%v, network=%s) failed: %w", port, network, err)
+			e.logf("UpdateMagicsockPort(port=%v, network=%s) failed: %v", port, network, err)
 		}
 	}
 	magicsockOpts := magicsock.Options{

